packages/client/src/modals/SuccessModal.js

```diff
@@ -0,0 +1,138 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Web3Context } from "contexts/Web3";
+import { useClipboard } from "../hooks";
+import { Copy, ArrowUp, OpenNewTab, Close } from "../components/Svg";
+import { getFlowscanUrlForTransaction } from "utils";
+import { ASSET_TYPES } from "../constants/enums";
+import { COIN_TYPE_TO_META, CONTRACT_NAME_TO_COIN_TYPE } from "../constants/maps";
+
+const SuccessModal = ({ data, type, txID, onClose, safeName, safeAddress, safeOwners }) => {
+    const { getNFTReference } = useContext(Web3Context);
+    const clipboard = useClipboard();
+
+    const [imageURI, setImageURI] = useState();
+
+    useEffect(() => {
+        if (type === ASSET_TYPES.NFT) {
+            const getImageURL = async () => {
+                const result = await getNFTReference(data.recipientAddr, data.nftID);
+                setImageURI(result);
+            };
+            getImageURL().catch(console.error);
+        }
+    }, [type])
+
+    function getNFTName(collectionId) {
+        return collectionId.split(".")[2];
+    }
+
+    function getTokenName(vaultId) {
```

Comment:
we can group this part of the logic into a getTokenMeta util function and use it like this:
`const {name, icon} = getTokenMeta(vaultId)`

Reply:
Makes perfect sense, thanks!

Reply:
Fixed!